lib/pv_web_utils.c

```diff
@@ -0,0 +1,210 @@
+/*
+    Copyright 2023 Picovoice Inc.
+    You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
+    file accompanying this source.
+    Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+    an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+    specific language governing permissions and limitations under the License.
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define PV_API __attribute__((visibility("default")))
+
+typedef enum pv_web_utils_status {
+    SUCCESS = 0,
+    FAILURE
+} pv_web_utils_status_t;
+
+extern void pv_console_log_wasm(const char *ch);
+extern void pv_assert_wasm(int32_t expr, int32_t line, const char *file_name);
+extern double pv_time_wasm(void);
+extern void pv_https_request_wasm(
+        const char *http_method,
+        const char *server_name,
+        const char *endpoint,
+        const char *header,
+        const char *body,
+        int32_t timeout_msec,
+        void **response,
+        size_t *response_size,
+        int32_t *response_code);
+extern void pv_get_browser_info(char **browser_info);
+extern void pv_get_origin_info(char **origin_info);
+extern void pv_file_open_wasm(void *f, const char *path, const char *mode, int32_t *status);
+extern void pv_file_close_wasm(void *f, int32_t *status);
+extern void pv_file_write_wasm(void *f, const void *content, size_t size, size_t count, size_t *num_write);
+extern void pv_file_read_wasm(void *f, void *content, size_t size, size_t count, int32_t *num_read);
+extern void pv_file_seek_wasm(void *f, long int offset, int whence, int32_t *status);
+extern void pv_file_tell_wasm(void *f, long *offset);
+extern void pv_file_remove_wasm(const char *path, int32_t *status);
```

Comment:
I'm not sure we should show off our internal function signatures. Or is this already public information? If not, let's keep it private and move this code to zoo-dev and deploy a library file to GitHub

Reply:
we can defer the signatures here: https://github.com/Picovoice/web-utils/blob/main/src/wasm.ts since they are the same. Also the reason this is put here is because Ali wanted this package to be isolated from the rest.